### PR TITLE
Plugins: Move alias support to plugin json (but still hardcoded)

### PR DESCRIPTION
**Pull request by ryantxu**
This moves the alias support to plugin.json, but still throws an error if someone besides the hardcoded plugins try to set it.

Ideally we can update the hosted grafana plugin submission logic so it won't accept anything with alias unless we hardcode it there 🤷 

but then release this with 10.1 -- that will mean that in the future there is a migration path for plugins like:
https://grafana.com/grafana/plugins/dlopes7-appdynamics-datasource/
https://grafana.com/grafana/plugins/alexanderzobnin-zabbix-app/

### CODE REVIEW

**Comment by ryantxu** [comment]
@DanCech -- thoughts on how to best support this in our plugin catalog? 

Perhaps we can update gcom api so it fails if someone submits a plugin.json with an "alias" field that is not on our hardcoded allow list?

**Comment by ryantxu** [comment]
@wbrowne -- i'll move this from "draft" to ready for review, only after there is some reasonable gate in the plugin submission process

**Comment by DanCech** [comment]
One option might be to only check for alias if the plugin is signed by the grafana org, which would mean that we don't need to hardcode a list of plugins or do any enforcement when validating plugin json.  It would be easy enough to add validation for plugin.json as well.

**Comment by ryantxu** [comment]
@DanCech -- I think we can remove the hardcoded stuff in the grafana code base -- the problem is that we have nothing preventing us from signing user submitted plugin.json files that include an alias.  If we are confident that we do not accidentally allow that -- then we can remove it from the software.

File: pkg/plugins/plugins.go [hunk 1]
```diff
@@ -166,10 +164,18 @@ func ReadPluginJSON(reader io.Reader) (JSONData, error) {
...
+		fallthrough
+	case "annolist":
+		fallthrough
+	case "debug":
+		if plugin.Alias == "" {
+			return plugin, fmt.Errorf("expected alias to be set")
+		}
+	default:
+		if plugin.Alias != "" {
+			return plugin, ErrUnsupportedAlias
```
Comment (ryantxu) [inline pkg/plugins/plugins.go:177]: @wbrowne 🤔 actually this is the only condition that needs to stay until we have something upstream in our publish pipeline to block it.  Are you comfortable moving forward with this, then we open a new PR to remove this error after we are confident we won't accidentally publish plugins with alias?

**Review by tolzhabayev — CHANGES_REQUESTED** [review]
let's sync on the plan here.

**Pull request closed without merging**